### PR TITLE
fix bug in AddEmptyLists where lub sort cannot be determined

### PR DESCRIPTION
**Pull request by dwightguth**
Previously we were getting an error in certain sort graphs saying that the least upper bound sort could not be determined, even though only one lub sort could actually appear in that location based on its expected sort. Here we correctly filter the lub by the expected sort, which we were not previously doing.

### CODE REVIEW

**Review by traiansf — APPROVED** [review]
It looks reasonable to me

File: k-distribution/tests/regression-new/lub/test.k [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  syntax PtrValue ::= SymLoc
+  syntax SymLoc
+  syntax IntPtr ::= EncodedPtr
+  syntax BitValue ::= EncodedPtr
+  syntax Encodable ::= PtrValue
+  syntax BitValueOrEncodable ::= BitValue | Encodable
+
+  syntax EncodedPtr ::= encodedPtr(PtrValue)
+  syntax Bool ::= sameBase(IntPtr, IntPtr) [function]
+  rule sameBase(_, (encodedPtr(S:SymLoc) => S))
```
Comment (radumereuta) [inline k-distribution/tests/regression-new/lub/test.k:14]: I guess the rewrite sort here is IntPtr?

File: kernel/src/main/java/org/kframework/parser/concrete2kore/disambiguation/AddEmptyLists.java [hunk 2]
```diff
@@ -63,6 +63,24 @@ public AddEmptyLists(Module m, Sort expectedSort) {
         this.expectedSort = expectedSort;
     }
 
+    private Tuple2<Either<Set<KEMException>, Term>, Set<KEMException>> superApply(TermCons tc) {
```
Comment (radumereuta) [inline kernel/src/main/java/org/kframework/parser/concrete2kore/disambiguation/AddEmptyLists.java:70]: The name of this function looks funny to me. Could you at least add some documentation?

**Review by radumereuta — APPROVED** [review]
Is there an issue that should be referenced? How did you find this example?
If you add the documentation you can merge it directly.

**Comment by denis-bogdanas** [comment]
Fixes #1121

**Comment by denis-bogdanas** [comment]
Fixes #1121 
Tested.

**Pull request merged**